hal/optic_flow.h

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+ * Copyright (c) 2009-2014, MAV'RIC Development Team
+ * All rights reserved.
+ * 
+ * Redistribution and use in source and binary forms, with or without 
+ * modification, are permitted provided that the following conditions are met:
+ * 
+ * 1. Redistributions of source code must retain the above copyright notice, 
+ * this list of conditions and the following disclaimer.
+ * 
+ * 2. Redistributions in binary form must reproduce the above copyright notice, 
+ * this list of conditions and the following disclaimer in the documentation 
+ * and/or other materials provided with the distribution.
+ * 
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" 
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE 
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF 
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+ * POSSIBILITY OF SUCH DAMAGE.
+ ******************************************************************************/
+
+/*******************************************************************************
+ * \file optic_flow.h
+ * 
+ * \author MAV'RIC Team
+ * \author Julien Lecoeur
+ *   
+ * \brief Defines types for generic optic flow sensors
+ *
+ ******************************************************************************/
+
+
+#ifndef OPTIC_FLOW_H_
+#define OPTIC_FLOW_H_
+
+#ifdef __cplusplus
+	extern "C" {
+#endif
+
+#include <stdint.h>
+
+
+typedef struct 
+{
+	int16_t x;
+	int16_t y;	
+} raw_of_vector_t;
+
+
+typedef struct
+{
+	float x;
+	float y;
+} of_vector_t;
+
+
+/**
+ * \brief Direction at which a ROI is pointed
+ */
+typedef struct
+{
+	float azimuth;		///< azimuth in radians
+	float elevation;	///< elevation in radians
+} viewing_direction_t;
+
+
+/**
+ * \brief Link between gyro rates and rotational optic flow
+ * 
+ * OF_rotation = A * gyro
+ * 
+ * [ OF_x ] = 	[Arx	Apx		Ayx] * 	[ roll_rate ]
+ * [ OF_y ]		[Ary	Apy		Ayy]	[ pitch_rate]
+ * 										[ yaw_rate 	]
+ */
+typedef struct
+{
+	float Arx;
+	float Apx;
+	float Ayx;
+	float Ary;
+	float Apy;
+	float Ayy;
+} derotation_matrix_t;
+
```

Comment:
@jlecoeur  
Could you add the definition of each component of the matrix as it should be done in any struct, please ?